src/components/course/data/utils.jsx

```diff
@@ -407,10 +409,344 @@ export function shortenString(str, maxLength, suffix, separator = ' ') {
+        block
+      >
+        Contact administrator
+        <span className="sr-only">for help</span>
+      </Button>
+    );
+  }
+
+  return null;
+};
+
+export const getMissingApplicableSubsidyReason = ({
+  enterpriseAdminUsers,
+  catalogsWithCourse,
+  couponCodes,
+  couponsOverview,
+  customerAgreementConfig,
+  subscriptionLicense,
+  containsContentItems,
+  missingSubsidyAccessPolicyReason,
+}) => {
+  // Default disabled enrollment reason, assumes enterprise customer does not have any administrator users.
+  let reasonType = DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY_NO_ADMINS;
+  let userMessage;
+  const hasEnterpriseAdminUsers = enterpriseAdminUsers?.length > 0;
+
+  // If there are admin users, change `reasonType` to use the
+  // `DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY` message.
+  if (hasEnterpriseAdminUsers) {
+    reasonType = DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY;
+  }
+
+  const couponCodesDisabledEnrollmentReasonType = getCouponCodesDisabledEnrollmentReasonType({
+    catalogsWithCourse,
+    couponCodes,
+    couponsOverview,
+    hasEnterpriseAdminUsers,
+  });
+  const subscriptionsDisabledEnrollmentReasonType = getSubscriptionDisabledEnrollmentReasonType({
+    customerAgreementConfig,
+    catalogsWithCourse,
+    subscriptionLicense,
+    hasEnterpriseAdminUsers,
+  });
+  const enterpriseOffersDisabledEnrollmentReasonType = getEnterpriseOffersDisabledEnrollmentReasonType({
+    enterpriseOffers: customerAgreementConfig?.enterpriseOffers,
+    catalogsWithCourse,
+  });
+  const learnerCreditDisabledEnrollmentReasonType = getLearnerCreditDisabledEnrollResponse({
+    redeemableSubsidyAccessPolicy: missingSubsidyAccessPolicyReason,
+    coursePrice: customerAgreementConfig?.price,
+  });
+
+  /**
+   * Prioritize the following order of disabled enrollment reasons:
+   * 1. Course not in catalog
+   * 2. Subscriptions related disabled enrollment reason
+   * 3. Coupon codes related disabled enrollment reason
+   * 4. Learner Credit related disabled enrollment reason.
+   * 5. Enterprise offers related disabled enrollment reason
+   */
+  if (enterpriseOffersDisabledEnrollmentReasonType) {
+    reasonType = enterpriseOffersDisabledEnrollmentReasonType;
+  }
+  if (missingSubsidyAccessPolicyReason) {
+    reasonType = missingSubsidyAccessPolicyReason.reason;
+    userMessage = missingSubsidyAccessPolicyReason.userMessage;
+  }
+  if (couponCodesDisabledEnrollmentReasonType) {
+    reasonType = couponCodesDisabledEnrollmentReasonType;
+  }
+  if (subscriptionsDisabledEnrollmentReasonType) {
+    reasonType = subscriptionsDisabledEnrollmentReasonType;
+  }
+  if (learnerCreditDisabledEnrollmentReasonType) {
+    reasonType = learnerCreditDisabledEnrollmentReasonType;
+  }
+  if (!containsContentItems) {
+    reasonType = DISABLED_ENROLL_REASON_TYPES.CONTENT_NOT_IN_CATALOG;
+  }
+  return {
+    reason: reasonType,
+    userMessage: userMessage ? reasonType : DISABLED_ENROLL_USER_MESSAGES[reasonType],
```

Comment:
The `userMessage` value should be a human-friendly string that explains why enrollment is disabled for the given `reasonType`. The `reason` attribute should be the only one returning the `reasonType`. 

It doesn't really make much sense to return the `reasonType` as the `userMessage` as the more programmatic slug for `reasonType` would be displayed instead of an actual English sentence (e.g., `policy_expired`, `policy_spend_limit_reached`, [etc.](https://github.com/openedx/enterprise-access/blob/46074dc25d80ce403baf47a9dabd2efaa7ed90ee/enterprise_access/apps/subsidy_access_policy/constants.py#L98-L105)).

Should this be updated to something more like the following such that it returns the given `userMessage` from the API, when appropriate?:

```javascript
return {
  reason: reasonType,
  userMessage: userMessage || DISABLED_ENROLL_USER_MESSAGES[reasonType],
}
```